java/src/jmri/implementation/AbstractTurnout.java

```diff
@@ -629,10 +643,17 @@ public String getDecoderName() {
         return _decoderName;
     }
 
-    /** {@inheritDoc} */
+    /** 
+     * {@inheritDoc} 
+     * On change, fires Property Change "decoderNameChange".
+     */
     @Override
-    public void setDecoderName(String decoderName) {
-        _decoderName = decoderName;
+    public void setDecoderName(final String decoderName) {
+        if (!(java.util.Objects.equals(_decoderName, decoderName))) {
```

Comment:
The code will bli slighter easier to read if you import java.util.Objects to this class and changes this to:
`if (!(Objects.equals(_decoderName, decoderName))) {`

You don't need to change this, and I wouldn't mention it if I hadn't had other comments as well.

Reply:
No problem, I've traditionally left out imports if it's just a a few uses, however I agree it's much easier to read :-)